tools/find_deprecated.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""List deprecated decorators."""
+
+from pathlib import Path
+from collections import OrderedDict, defaultdict
+import ast
+from datetime import datetime
+import requests
+
+
+class Deprecation:
+    """
+    Deprecation node, representing a single deprecation decorator.
+
+    Args:
+        filename: where is the deprecation.
+        decorator_node: AST node of the decorator call.
+        func_node: AST node of the decorated call.
+    """
+
+    def __init__(self, filename, decorator_node, func_node):
+        self.filename = filename
+        self.decorator_node = decorator_node
+        self.func_node = func_node
+        self._since = None
+
+    @property
+    def type(self):
```

Comment:
It's typically better to avoid `type` in Python because `type()` is a builtin and Python is permissive with letting you shadow the name, which can cause issues.

I usually go with `kind` instead as a synonym.